var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -93,7 +93,9 @@ class Gcc(AutotoolsPackage):
     depends_on('mpfr@2.4.2:')
     depends_on('mpc@0.8.1:', when='@4.5:')
     depends_on('isl@0.14', when='@5:5.9')
-    depends_on('isl@0.15:', when='@6:')
+    depends_on('isl@0.15:0.19', when='@6:8.9')
+    # https://gcc.gnu.org/bugzilla/show_bug.cgi?id=86724
```

Comment:
@ax3l 

IMO a short description of the issue is worth including with the link (so the user doesn't have to follow the link to know why it's there).

Also, that link mentions "Fixed for GCC 7.4+.". Does your experience suggest otherwise?

Upper bounds on dependency versions are fine but come with their own "risk": as new package versions are released, they may start to depend on later versions of dependencies, which can get out of sync if no one touches this package. That is unlikely for the Spack GCC package though.

Potentially the "safest" approach is to only apply the dependency version constraint up to the latest version of GCC. That way, when someone adds a new GCC version, we can decide whether to propagate the constraint along to that version.

In this case though, the `when=@9:` constraint appears to be speculative for future GCC versions (since the latest Spack version is 8.2.0). IMO this should be a suggestion in a comment vs. a hard constraint in the package.

Reply:
> IMO a short description of the issue is worth including with the link

I agree, it's just a compile error.

> Upper bounds on dependency versions ...

Come at the risk to narrow down possible combinations, but mitigate the risk to introduce failing builds that users face immediately, just because someone updated a quite fundamental c library in spack. This problem will persist as long as we follow the "always newest" philosophy and hits all our users directly.

> Potentially the "safest" approach is to only apply the dependency version constraint up to the latest version of GCC. That way, when someone adds a new GCC version, we can decide whether to propagate the constraint along to that version.

Fully agree! I second to do it that way! :)

> that link mentions "Fixed for GCC 7.4+.".

uh, yes there are bugfix releases that backported the fix. So you suggest to write `depends_on('isl@0.15:0.20', when='@8.2.1:,7.3.1:7.99.99)`? I found that too verbose for the only effect to increase the dependency range for an older release (with no immanent benefit since 0.15 will do as well) but can do.

> `when=@9:` constraint appears to be speculative

agreed, thx!

Reply:
> there are bugfix releases that backported the fix. So you suggest to write `depends_on('isl@0.15:0.20', when='@8.2.1:,7.3.1:7.99.99)`

Originally I assumed you could just say `depends_on('isl@0.15:0.20', when='@7.3.1:)`, not realizing there were `8.2.x` versions before `8.2.1` without the patch.

So to be clear this part is fine: I'm OK with using simpler constraints even if they are stricter than necessary (there's a tradeoff there but I think it's fine in this case).

Reply:
Actually, I was intiially trying to fix a 7.3 and 5.3/4 issue users reported as well, but could not reproduce it anymore.

Now, we have a report in #6902 that also needs an upper limit for those